### PR TITLE
add getAccountState

### PR DESCRIPTION
**Pull request by superboyiii**

### CODE REVIEW

**Comment by superboyiii** [comment]
Test result
```
neo> get accountstate Nb5PpPX7TtwFjVrzghRgGQtzuQNM6pVLGD
Invoking script with: 'DBSmUjEciaHgpVP4AibxfrGvQYMi4BHAHwwPZ2V0QWNjb3VudFN0YXRlDBT1Y+pAvCg9TQ4FxI6jBbPyoHNA70FifVtS'
VM State: HALT
Gas Consumed: 0.0202833
Result Stack: [{"type":"Struct","value":[{"type":"Integer","value":"5000000"},{"type":"Integer","value":"137"},{"type":"ByteString","value":"AwCbdUDhDyVi5f2PrJ6uwlFmpYsm5BI0j/WoaSe/rCKi"}]}]

Voted:
NXgxx4sGFB9fdonCdjpweiiPfypMpppDuj      5000000         block: 137
```

File: neo-cli/CLI/MainService.Vote.cs [hunk 1]
```diff
@@ -210,5 +212,29 @@ private void OnGetNextBlockValidatorsCommand()
                 }
             }
         }
+
+        /// <summary>
+        /// Process "get next validators"
```
Comment (shargon) [inline neo-cli/CLI/MainService.Vote.cs]: Message

File: neo-cli/CLI/MainService.Vote.cs [hunk 2]
```diff
@@ -210,5 +212,29 @@ private void OnGetNextBlockValidatorsCommand()
...
+            if (!OnInvokeWithResult(NativeContract.NEO.Hash, "getAccountState", out StackItem result, null, new JArray(arg))) return;
+
+            var resJArray = (VM.Types.Array)result;
+            if (resJArray.Count > 0)
+            {
+                Console.WriteLine();
+                Console.WriteLine("Voted:");
+                var publickey = ECPoint.Parse(((ByteString)resJArray?[2])?.GetSpan().ToHexString(), ECCurve.Secp256r1);
+                Console.Write(Contract.CreateSignatureRedeemScript(publickey).ToScriptHash().ToAddress(NeoSystem.Settings.AddressVersion) + "\t");
+                Console.Write(new BigDecimal(((Integer)resJArray?[0]).GetInteger(), NativeContract.NEO.Decimals) + "\t\t");
```
Comment (shargon) [inline neo-cli/CLI/MainService.Vote.cs:235]: One line per value?

File: neo-cli/CLI/MainService.Vote.cs (hunk 1 above)
Comment (superboyiii) [reply neo-cli/CLI/MainService.Vote.cs]: > Message

fixed

File: neo-cli/CLI/MainService.Vote.cs [hunk 3]
```diff
@@ -210,5 +212,29 @@ private void OnGetNextBlockValidatorsCommand()
                 }
             }
         }
+
+        /// <summary>
+        /// Process "get accountstate"
+        /// </summary>
+        [ConsoleCommand("get accountstate", Category = "Vote Commands")]
```
[the remark this answers is not in the document]
Comment (superboyiii) [reply neo-cli/CLI/MainService.Vote.cs:219]: > category?

Which you prefer?

**Comment by superboyiii** [comment]
Result now:
```
neo> get accountstate Nb5PpPX7TtwFjVrzghRgGQtzuQNM6pVLGD
Invoking script with: 'DBSmUjEciaHgpVP4AibxfrGvQYMi4BHAHwwPZ2V0QWNjb3VudFN0YXRlDBT1Y+pAvCg9TQ4FxI6jBbPyoHNA70FifVtS'
VM State: HALT
Gas Consumed: 0.0202833
Result Stack: [{"type":"Struct","value":[{"type":"Integer","value":"5000000"},{"type":"Integer","value":"137"},{"type":"ByteString","value":"AwCbdUDhDyVi5f2PrJ6uwlFmpYsm5BI0j/WoaSe/rCKi"}]}]

Voted: NXgxx4sGFB9fdonCdjpweiiPfypMpppDuj
Amount: 5000000
Block: 137
```

File: neo-cli/CLI/MainService.Vote.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (shargon) [reply neo-cli/CLI/MainService.Vote.cs:219]: I am not sure, but it's not only for voting, we can change it later

**Review by shargon — APPROVED** [review]

**Pull request merged**